lib/services/ftp.js

```diff
@@ -343,13 +343,48 @@ exports.rm = function (path, request, response, next, cbk) {
       exports.errNotLoggedIn(cbk);
     }
     else {
-      ftpClientsArray[request.session.ftpClientId].delete(path, function (err) {
-        if (err){
-          console.error(err);
-          cbk({success:false, message: err});
-        }
-        else
-          cbk({success:true});
+      path = path.replace(/\/+$/,''); // remove last slash on directories
+      var parentDir = path.replace(/^(.*)\/[^\/]+$/,'$1');
+      ftpClientsArray[request.session.ftpClientId].list(parentDir, function(err, filesArray) {
+          if(err)
+          {
+            cbk({success:false, message: err});
+          }
+          else
+          {
+            var filename = path.replace(/^.*\//,'');
```

Comment:
Since `path` is the dirName, aren't you erasing the all string?
Once again, it will be clearer to use native method like [basename()](https://devdocs.io/node~4_lts/path#path_path_basename_p_ext)